scripts/normalize-by-median.py

```diff
@@ -70,31 +66,42 @@ def WithDiagnostics(ifile, batch_size, fp, paired, norm):
                           perc=int(100. - discarded / float(total) * 100.)),
                   file=sys.stderr)
 
-            print('... in file ' + input_filename, file=sys.stderr)
+            print('... in file ' + ifilename, file=sys.stderr)
 
-            if report_fp:
+            if fp:
                 print(total + " " + total - discarded + " " +
                       1. - (discarded / float(total)), file=fp)
                 report_fp.flush()
 
-        # If in paired mode, check that the reads are properly interleaved
-        if paired:
-            if not check_is_pair(batch[0], batch[1]):
-                raise IOError('Error: Improperly interleaved pairs \
-                    {b0} {b1}'.format(b0=batch[0].name, b1=batch[1].name))
+        yield index, is_paired, read0, read1
 
-        yield batch
+    # per file diagnostic output
+    if norm.total == 0 and norm.discarded == 0:
```

Comment:
(if you were unsure, you could put in an assertion:

    if norm.total == 0:
        assert norm.discarded == 0
        ...